utils/search/search_forum.py

```diff
@@ -17,73 +17,84 @@
 # Authors:
 #     See AUTHORS file.
 #
+import logging
 
-from solr import Solr, SolrException
 from django.conf import settings
-import logging
+
+import forum.models
+from solr import Solr, SolrException
+from utils.text import remove_control_chars
 
 logger = logging.getLogger("search")
 
-def convert_to_solr_document(post):
-    #logger.info("creating solr XML from forum post %d" % post.id)
-    document = {}
 
-    document["id"] = post.id
-    document["thread_id"] = post.thread.id
-    document["thread_title"] = post.thread.title
-    document["thread_author"] = post.thread.author
-    document["thread_created"] = post.thread.created
+def convert_to_solr_document(post):
+    document = {
+        "id": post.id,
+        "thread_id": post.thread.id,
+        "thread_title": remove_control_chars(post.thread.title),
+        "thread_author": post.thread.author,
+        "thread_created": post.thread.created,
 
-    document["forum_name"] = post.thread.forum.name
-    document["forum_name_slug"] = post.thread.forum.name_slug
+        "forum_name": post.thread.forum.name,
+        "forum_name_slug": post.thread.forum.name_slug,
 
-    document["post_author"] = post.author
-    document["post_created"] = post.created
-    document["post_body"] = post.body
+        "post_author": post.author,
+        "post_created": post.created,
+        "post_body": remove_control_chars(post.body),
 
-    document["num_posts"] = post.thread.num_posts
-    document["has_posts"] = False if post.thread.num_posts == 0 else True
+        "num_posts": post.thread.num_posts,
+        "has_posts": False if post.thread.num_posts == 0 else True
+    }
 
     return document
 
 
-def add_post_to_solr(post):
-    logger.info("adding single forum post to solr index")
+def send_posts_to_solr(posts):
+    logger.info("adding forum posts to solr index")
+    logger.info("creating XML")
+    documents = [convert_to_solr_document(p) for p in posts]
+
     try:
-        Solr(settings.SOLR_FORUM_URL).add([convert_to_solr_document(post)])
+        logger.info("posting to Solr")
+        solr = Solr(settings.SOLR_FORUM_URL)
+
+        solr.add(documents)
+
+        solr.commit()
     except SolrException as e:
-        logger.error("failed to add forum post %d to solr index, reason: %s" % (post.id, str(e)))
+        logger.error("failed to add posts to solr index, reason: %s" % str(e))
+    logger.info("done")
 
 
-def add_posts_to_solr(posts):
-    logger.info("adding multiple forum posts to solr index")
-    solr = Solr(settings.SOLR_FORUM_URL, auto_commit=False)
+def add_post_to_solr(post_id):
+    """Add a forum post to solr
+    Arguments:
+        post_id (int): ID of a post object"""
```

Comment:
wrong docstring format, although I'm starting to like this one better because the types are more readable... Maybe we should switch.

Reply:
yeah, we had this conversation. Tell me which one you prefer and I'll add it to the programming guidelines.

Reply:
we decided to use the reStructuredText/Sphinx, but I like how parameter types are shown here, it is more readable. maybe we should use this (google) one.